### ISSUE TITLE
ChartSerializer race condition

### ISSUE THREAD

**Issue by matt-entrekin** [apexcharts/Blazor-ApexCharts#241]
I have a Blazor Server application with multiple charts that call `ApexChart.UpdateOptionsAsync` in `OnAfterRenderAsync`. In certain debug situations I encountered an error where connecting two clients at the same time can lead to a race condition with the dictionary in `ChartSerializer`.

This is the error

```
System.AggregateException: One or more errors occurred. (An item with the same key has already been added. Key: <redacted_type_name>)
 ---> System.ArgumentException: An item with the same key has already been added. Key: <redacted_type_name>
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at BlazorApexCharts.ChartSerializer.GetOptions[TItem]()
   at ApexCharts.ApexChart`1.Serialize(Object data)
   at ApexCharts.ApexChart`1.UpdateOptionsAsync(Boolean redrawPaths, Boolean animate, Boolean updateSyncedCharts)
   at <redacted_call_site>.OnAfterRenderAsync(Boolean firstRender) in <redacted_file_name>
   --- End of inner exception stack trace ---
```

I think this is limited to Blazor Server because of concurrency. In theory, it shouldn't be a problem in Blazor WASM. It may suffice to replace `Dictionary<,>` with `ConcurrentDictionary<,>` in `ChartSerializer`.

**Comment by haas-daniel** [comment]
I have the same problem in Blazor Server.

My workaround:
I call `new ChartSerializer().GetOptions<redacted_type_name>();` manually at startup (e. x. in program.cs). This way the options are in the dictionary already before any chart is rendered.

**Comment by joadan** [comment]
fixed in 0.9.22
